src/models/state/mod.rs

```diff
@@ -1618,6 +1620,21 @@ impl GlobalState {
             ));
         }
 
+        let mut pow_witnesses: Vec<(BlockHeader, BlockPoWWitness)> = vec![];
+        let mut block_hash = tip.hash();
+        while pow_witnesses.len() < SYNC_CHALLENGE_POW_WITNESS_LENGTH {
+            let (block_header, pow_witness) = self
+                .chain
+                .archival_state()
+                .block_header_with_pow_witness(block_hash)
+                .await
+                .unwrap_or_else(|| {
+                    panic!("Pow-witness for block with hash {block_hash} must exist")
```

Comment:
There is something missing from this statement that makes its truth obvious. Is it because archival nodes always store the proof-of-work witnesses for all blocks stored?